core/modules/replica/Controller.h

```diff
@@ -0,0 +1,687 @@
+     * to the worker-side service and return detailed info on all known requests.
+     *
+     * @param workerName - the name of a worker node where the service runs
+     * @param onFinish   - a callback function to be called upon completion of the operation
+     * @param jobId      - an optional identifier of a job issed the request
+     * @param requestExpirationIvalSec
+     *                   - an optional parameter (if differs from 0)
+     *                     allowing to override the default value of the corresponding
+     *                     parameter from the Configuration.
+     *
+     * @return a pointer to the request
+     */
+    ServiceDrainRequest_pointer drainWorkerService(
+                                        std::string const& workerName,
+                                        ServiceDrainRequest_callback_type onFinish=nullptr,
+                                        std::string const& jobId="",
+                                        unsigned int requestExpirationIvalSec=0);
+
+    /**
+     * Return requests of a specific type
+     *
+     * @param requests - a collection to be populated with requests
+     */
+    template <class REQUEST_TYPE>
+    void requestsOfType(std::vector<typename REQUEST_TYPE::pointer>& requests) const {
+        std::lock_guard<std::mutex> lock(_mtx);
+        requests.clear();
+        for (auto&& itr: _registry)
+            if (typename REQUEST_TYPE::pointer ptr =
+                std::dynamic_pointer_cast<REQUEST_TYPE>(itr.second->request())) {
+                requests.push_back(ptr);
+            }
+    }
+
+    /**
+     * Return the number of requests of a specific type
+     */
+    template <class REQUEST_TYPE>
+    size_t numRequestsOfType() const {
+        std::lock_guard<std::mutex> lock(_mtx);
+        size_t result {0};
+        for (auto&& itr: _registry) {
+            if (typename REQUEST_TYPE::pointer request =
+                std::dynamic_pointer_cast<REQUEST_TYPE>(itr.second->request())) { ++result; }
+        }
+        return result;
+    }
+
+    /**
+     * Return the total number of requests of all kinds
+     */
+    size_t numActiveRequests() const;
+
+private:
+
+    /**
+     * Construct the server with the specified configuration.
+     *
+     * @param serviceProvider - for configuration, other services
+     */
+    explicit Controller(ServiceProvider::pointer const& serviceProvider);
+
+    /// @return the context string for debugging and diagnostic printouts
+    std::string context() const;
+
+    /**
+     * Finalize the completion of the request. This method will notify
+     * a requestor on the completion of the operation and it will also
+     * remove the request from the server's registry.
+     */
+    void finish(std::string const& id);
+
+    /**
+     * Make sure the server is runnning. Otherwise throw std::runtime_error.
+     */
+    void assertIsRunning() const;
+
+private:
```

Comment:
this `private` is redundant.

Reply:
This is true. However, I always use a separate "private" (etc.) for member functions and data members to avoid issue in case if the previous one gets removed or relaxed.